### ISSUE TITLE
Error processing page (javascript - Ajax) after insert Captcha value

### ISSUE THREAD

**Issue by ceceussj4** [HtmlUnit/htmlunit#228]
I'm using Htmlunit 2.43 with Spring. 
When I click the button and call Ajax JavaScript I see in log the 
 Re-synchronized call to https://www.licitacoes-e.com.br/aop/listar-lotes.aop?opcao=listarLotesPosCaptcha&numeroLicitacao=824678&numeroLoteInicial=0&numeroLoteFinal=0&question=68pwr and it's ok, this link open the correct Page, but the variable Page doesn't change with the value correct the link.

My Code:

    WebClient client = new WebClient(BrowserVersion.CHROME);

    CookieManager cookieManager = client.getCookieManager();
    cookieManager.setCookiesEnabled(true);
    client.setCookieManager(cookieManager);
    client.getOptions().setJavaScriptEnabled(true);
    client.getOptions().setThrowExceptionOnScriptError(false);
    client.getOptions().setCssEnabled(false);
    client.setAjaxController(new NicelyResynchronizingAjaxController());
    client.getOptions().setPopupBlockerEnabled(true);
    client.getOptions().setThrowExceptionOnFailingStatusCode(false);
    client.getOptions().setUseInsecureSSL(true);
    client.getOptions().setRedirectEnabled(true);
    .....
    HtmlTextInput inputCaptcha = 
      paginaAfterConsultaIdentificadorCaptcha.querySelector("input[name='questionCL']");
    inputCaptcha.setValueAttribute("4wh7m");

    paginaAfterConsultaIdentificadorCaptcha = ((HtmlButtonInput) 
    paginaAfterConsultaIdentificadorCaptcha.getElementById("botao_continuar")).click();
    client.waitForBackgroundJavaScript(10000);
    System.out.println(paginaAfterConsultaIdentificadorCaptcha.asXml());

     I already tried this too:
      for (int i = 0; i < 20; i++) {
            synchronized (paginaAfterConsultaIdentificadorCaptcha) {
                paginaAfterConsultaIdentificadorCaptcha.wait(10000);
            }
        }
       System.out.println(paginaAfterConsultaIdentificadorCaptcha.asXml());

In console log show me:

2020-09-14 15:39:45.201  INFO 2573 --- [pool-2-thread-1] .g.h.NicelyResynchronizingAjaxController : Re-synchronized call to https://www.licitacoes-e.com.br/aop/listar-lotes.aop?opcao=listarLotesPosCaptcha&numeroLicitacao=824678&numeroLoteInicial=0&numeroLoteFinal=0&question=4wh7m

2020-09-14 15:39:47.687  WARN 2573 --- [Client@2760b96d] c.g.htmlunit.html.HtmlScript             : Script is not JavaScript (type: 'true/text/javascript', language: ''). Skipping execution.

2020-09-14 15:39:47.687  WARN 2573 --- [Client@2760b96d] c.g.htmlunit.html.HtmlScript             : 
Script is not JavaScript (type: 'true/text/javascript', language: ''). Skipping execution.

**Comment by ceceussj4** [comment]
I realized that deleting the cookies (JSESSIONID) in Browser(FireFox) show the same page that the htmlunit after execute Ajax. But I put the cookie in code and the page doesn't return expected page.

**Comment by ceceussj4** [comment]
I realized too that the new page that is call by ajax has a code  
<script type="text/javascript">
	$(document).ready(function(){
		inicializaScripts();
	});
</script>

but not refer um jquery file. 
Will be that it can be the error?

**Comment by ceceussj4** [comment]
I believe that the error is when the ajax call open the link. This page needs the cookies, but I don't know. I tried everything. I putted the cookies in webClient, putted in header but doen't work. Really, I need help.

**Comment by rbri** [comment]
Not sure i got your point - is the page public available?

**Comment by rbri** [comment]
And please try to refresh your page variable (just in case the page got replaced by js) by doing something like

    client.waitForBackgroundJavaScript(10000);
    page = page.getEnclosingWindow().getEnclosedPage();

**Comment by ceceussj4** [comment]
First, thanks for answer!
This is a government page and I use HtmlUnit for get datas.
I tried you example code: 
client.waitForBackgroundJavaScript(10000); 
page = page.getEnclosingWindow().getEnclosedPage();

But sorry doen't worked!

This page use Cookie Session (JSESSIONID) when I deleted this cookie in browser (Chrome) happened the same error.

So, will be that the error is when send this cookie?

Other detail when I click in the button this function that is call:

<input id="botao_continuar" type="button" value="continuar" class="button col" onclick="javascript:consultarLotesLicitacaoVerificaCaptcha(824678,0,0,document.getElementById('inputCaptchaConsultarLotes').value);">

<script type="text/javascript">
function consultarLotesLicitacaoPosCaptcha(numeroLicitacao,numeroLoteInicial,numeroLoteFinal,valor) {	
		limparIds();	
		url = 'https://www.licitacoes-e.com.br/aop/listar-lotes.aop';
	    url += '?opcao=listarLotesPosCaptcha';
	    url += '&numeroLicitacao=' + numeroLicitacao;
	    url += '&numeroLoteInicial=' + numeroLoteInicial;
	    url += '&numeroLoteFinal=' + numeroLoteFinal;
	    url += '&question=' + valor;
	    link(url, {moldura:false, iframe:"#id_listagem"});
	} 
<script>.

This function process new page and show the result in div = #id_listagem;

I will install Charles Proxy to show if the cookies are going!

I'm sorry I know that my English is terrible and thanks again for help me!
